File: apps/web/app/(protected)/documents/feature/types/readProgress.ts
export type SaveReadProgressRequest = {
  documentId: number;
  tgtLang: string;
  currentPageNumber: number; // 1-indexed for backend
};

export type SaveReadProgressResponseDTO = {
  document_id: number;
  tgt_lang: string;
  current_page_number: number;
  completion_percent: number;
  total_pages: number;
  last_read_at: string | null;
  completed_at: string | null;
};

File: services/api/src/api/schemas/read_progress.py
from pydantic import BaseModel

class SaveReadProgressRequest(BaseModel):
    document_id: int
    tgt_lang: str
    current_page_number: int

class SaveReadProgressResponse(BaseModel):
    document_id: int
    tgt_lang: str
    current_page_number: int
    completion_percent: int
    total_pages: int
    last_read_at: str | None
    completed_at: str | None